pkg/proto/nbcontract/v1/README.md

```diff
@@ -46,4 +46,98 @@ This table is describing the all the AKSNodeConfig Fields converted to .go files
+
+Notes: In proto3, all variables are _optional_, not required. Thus indeed, `optional` label in proto3 doesn’t really mean it’s an _optional_ variable. It’s saying that it’s explicit presence. (I know it’s confusing). The concept of `optional` label is to distinguish between these 2 cases.
+-	A variable is not set, meaning assigned with no value. In case of no presence, it will be automatically assigned with the default value. Depending on different types of variables, `proto3` has different default values. Check more [here](https://protobuf.dev/programming-guides/proto3/#default) if interested.
+-	A variable is explicitly assigned with a value, which happens to be the default value.
+For some cases, knowing that the variable is not set is important. Then the feature owner can handle it with additional logic.
+For the best practice, if the feature doesn’t require distinguishing the 2 cases above, please don’t add an `optional` label. If it’s needed to distinguish between, please add an `optional` label.
+Nevertheless, it’s not a big harm to use `optional` even though it’s not needed. It’s just on the consumer side, you will need to either use the proto3 generated getters, which ensures non-nil value, or handle the nil value properly by yourself. But you may also want to let other people know that it could be nil value when they use the variable you added.
+
+
+## High level Steps
+1. Update corresponding .proto files to the data contract. Usually we start with `config.proto`.
+1. At `AgentBaker` root level, run command `make compile-proto-files` to compile the .proto files to .go files. At this point, we updated the data contract.
+1. Tell how VHD should react to this new variable by updaing the bootscripts as you do before. Basically you will be modifying shell scripts like `install-dependencies.sh`, `cse_install.sh`, `cse_helpers.sh`, etc. You may also want to add some unit tests to spec files like `cse_install_spec.sh`, `cse_helpers.sh` to find bugs earlier.
+1. On the VHD side, we are still invoking the bootstrap scripts under the hood. So we will need to update cse_cmd.sh.gtpl to set the environment variables of the CSE trigger command. Note: Node SIG is working on migrating all scripts to managable Go binary. Before it's done, the bootstrap scripts will still be used.
+1. Set default values for your variables, if the existing defaulting provided by `proto3` doesn't fit your purpose. For example, if a bool variable is not set, `proto` will default it to `false`. However, if you want to default it to `true`, then you can set your own default function. `getDisableSSH` in `cse_cmd.sh.gtpl` is 1 example.
+
+## Detailed steps with example
+Example: IMDSRescrtionConfig
```

Comment:
Might be nice to include a link to this PR as well

Reply:
added